main.go

```diff
@@ -121,6 +130,49 @@ func main() {
 					os.Exit(1)
 				}
 			}
+		} else if module, _ := tfconfig.LoadModule(dir); len(module.RequiredCore) >= 1 && len(args) == 0 { //if there is required_version defined in TF code, and no commmand line arguments
+			tfversion := ""
+
+			// we skip duplicated definitions and use only first one
+			tfconstraint := module.RequiredCore[0]
+			tflist, _ := lib.GetTFList(hashiURL, true)
+			fmt.Printf("Reading required version from terraform code, constraint: %s\n", tfconstraint)
+			//fmt.Printf("Available versions: %#v\n", tflist)
+
+			c, err := semver.NewConstraint(tfconstraint)
+			if err != nil {
+				fmt.Println("Error parsing constraint:", err)
+				return
+			}
+			vs := make([]*semver.Version, len(tflist))
+			for i, r := range tflist {
+				v, err := semver.NewVersion(r)
+				if err != nil {
+					fmt.Printf("Error parsing version: %s", err)
+				}
+
+				vs[i] = v
+			}
+
+			sort.Sort(semver.Collection(vs))
```

Comment:
If you would use https://golang.org/pkg/sort/#Reverse you could later iterate "in normal way" over elements, instead of using this index based notation.